### PR TITLE
fix sporadic timer ObjectDisposedException

### PR DESCRIPTION
**Pull request by cristipogacean**
* validate if the timer was disposed before using it
* tuned _healthCheckManager.IsReady for better reporting

### CODE REVIEW

**Review by dacolgit — DISMISSED** [review]
:shipit:

**Review by dacolgit — APPROVED** [review]

File: common/src/Microsoft.Azure.IIoT.Agent.Framework/src/Agent/Default/WorkerSupervisor.cs [hunk 1]
```diff
@@ -102,47 +104,53 @@ public class WorkerSupervisor : IWorkerSupervisor, IDisposable {
...
-            await worker.Key.StopAsync();
+            await worker.Key?.StopAsync();
             worker.Value?.Dispose();
         }
 
         /// <inheritdoc/>
         public void Dispose() {
             if (_ensureWorkerRunningTimer != null) {
-                _ensureWorkerRunningTimer.Stop();
+                _ensureWorkerRunningTimer.Enabled = false;
```
Comment (karok2m) [inline common/src/Microsoft.Azure.IIoT.Agent.Framework/src/Agent/Default/WorkerSupervisor.cs:119]: Why is there no need to call `_ensureWorkerRunningTimer.Stop()` now ?

Comment (karok2m) [reply common/src/Microsoft.Azure.IIoT.Agent.Framework/src/Agent/Default/WorkerSupervisor.cs:119]: Ah, because `Stop()` just sets `Enabled` to `false`.

**Review by karok2m — APPROVED** [review]

File: common/src/Microsoft.Azure.IIoT.Agent.Framework/src/Agent/Default/WorkerSupervisor.cs (hunk 1 above)
Comment (cristipogacean) [reply common/src/Microsoft.Azure.IIoT.Agent.Framework/src/Agent/Default/WorkerSupervisor.cs:119]: _ensureWorkerRunningTimer.Stop() is actually the same as: _ensureWorkerRunningTimer.Enabled = false. I made this change consistently through this class

**Pull request merged**